src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -178,43 +173,35 @@ public SqlConnection(string connectionString, SqlCredential credential) : this()
                 {
                     throw ADP.InvalidMixedArgumentOfSecureCredentialAndIntegratedSecurity();
                 }
-
-                if (UsesContextConnection(connectionOptions))
+                else if (UsesContextConnection(connectionOptions))
```

Comment:
nit: this one is one I've butted heads with people in the past. Technically, the else is redundant here since if the condition is triggered, we exit the method. In my IDE, I get notices about them, so *I* usually try to avoid them. But I admit, it is tidier to have the else-if chain. Though it would be even better if this could be written as a switch block :/